### PR TITLE
Merge latest master and fix skips

### PR DESCRIPTION
**Pull request by liach**

### CODE REVIEW

**Comment by liach** [comment]
Fixed conflict.

File: src/java.base/share/classes/jdk/internal/classfile/impl/UnboundAttribute.java [hunk 1]
```diff
@@ -849,11 +849,11 @@ public AdHocAttribute(AttributeMapper<T> mapper) {
         @Override
         public void writeTo(BufWriterImpl b) {
             b.writeIndex(b.constantPool().utf8Entry(mapper.name()));
-            b.writeInt(0);
+            int lengthIndex = b.skip(4);
             int start = b.size();
             writeBody(b);
             int written = b.size() - start;
-            b.patchInt(start - 4, written);
+            b.patchInt(lengthIndex, written);
```
Comment (wenshao) [inline src/java.base/share/classes/jdk/internal/classfile/impl/UnboundAttribute.java:855]: There is one more variable here. The original start - 4 is fine.

Comment (liach) [reply src/java.base/share/classes/jdk/internal/classfile/impl/UnboundAttribute.java:855]: Can I remove `start`, and calculate `written` like this?
```java
int written = b.size() - lengthIndex - 4;
```

Comment (liach) [reply src/java.base/share/classes/jdk/internal/classfile/impl/UnboundAttribute.java:855]: Removed redundant variables.

**Pull request merged**